readme.txt
=== Sync Holded for WooCommerce or Easy Digital Downloads ===
Contributors: closemarketing, davidperez, sacrajaimez, freemius
Tags: holded, woocommerce
Donate link: https://close.marketing/go/donate/
Requires at least: 4.0
Requires PHP: 5.6
Tested up to: 5.8.2
Stable tag: 1.3
Version: 1.3

Syncs Products and data from Holded software to WooCommerce or Easy Digital Downloads.

== Description ==

This plugin allows you to import simple products from Holded to WooCommerce. 

It creates a new menu in WooCommerce > Products > Import products from Holded.

You can import simple products, and it will create new products if it does not find the SKU code from your WooCommerce. If the SKU exists, it will import all data except title and description from the product. The stock will be imported as well.

¡We have a Premium version!
These are the features:
- Import categories from Holded.
- Import attributes as brands or others.
- Import variable products.
- Automate the syncronization.
- Sync Orders to Holded.
- Import pack products from Holded.

[You could buy it here](https://checkout.freemius.com/mode/dialog/plugin/5133/plan/8469/)

Made by [Closemarketing](https://close.marketing/)

== Installation ==

Extract the zip file and just drop the contents in the wp-content/plugins/ directory of your
WordPress installation and then activate the Plugin from Plugins page.

== Developers ==
[Official Repository GitHub](https://github.com/closemarketing/import-holded-products-woocommerce)

== Changelog ==
= 1.3 =
*   Sync orders to Holded (Premium) automatically and force manually for past orders.
*   Sync Pack products to Holded (Premium).
*   Fix: Attributes duplicated in variation product not imported.
*   Fix: Categories not imported in simple products.

= 1.2 =
*   Automate your syncronization! (Premium).
*   Option email when is finished (Premium).
*   Fix sku saved for EDD.
*   Better metavalue search for SKU.
*   Fix Holded Pagination (thanks to itSerra).
*   Fix SKU variation (thanks to itSerra).

= Earlier versions =

For the changelog of earlier versions, please refer to the separate changelog.txt file.

== Links ==
*	[Closemarketing](https://close.marketing/)
*	[Closemarketing plugins](https://profiles.wordpress.org/closemarketing/#content-plugins)


changelog.txt
= Changelog archive =

This is an archive of older changelog entries. Most recent entries are maintained in readme.txt

= 1.2 =
*   Automate your syncronization! (Premium).
*   Option email when is finished  (Premium).
*   Fix sku saved for EDD.
*   Better metavalue search for SKU.
*   Fix Holded Pagination (thanks to itSerra).
*   Fix SKU variation (thanks to itSerra).


= 1.1 =
*   Fix bug not import files full list of products.
*   Import Simple products to Easy Digital Downloads.
*   [Premium] Import image files from Holded for WooCommerce and EDD.

= 1.0 =
*   Improved log syncronization.
*   Better consitency between API Holded and WooCommerce.
*   Refactored syncronization engine.
*   Implemented Premium process.

Premium Version:
*   Import Categories nested.
*   Import variable products.

= 0.2 =
*   Implemented library for statistics.
*   First implementation of premium product.

= 0.1 =
*   First released.

import-holded-products-woocommerce.php
<?php

/**
 * Plugin Name: Sync Ecommerce Holded
 * Plugin URI: https://www.closemarketing.es
 * Description: Syncs Products and data from Holded to WooCommerce or Easy Digital Downloads.
 * Author: closemarketing
 * Author URI: https://www.closemarketing.es/
 * Version: 1.3
 * WC requires at least: 5.0
 * WC tested up to: 5.8.2
 *
 * @package WordPress
 * Text Domain: import-holded-products-woocommerce
 * Domain Path: /languages
 * License: GNU General Public License version 3.0
 * License URI: http://www.gnu.org/licenses/gpl-3.0.html
 *
 */
defined( 'ABSPATH' ) || exit;
define( 'WCPIMH_VERSION', '1.3' );
define( 'WCPIMH_ECOMMERCE', array( 'woocommerce', 'edd' ) );
define( 'WCPIMH_TABLE_SYNC', 'wcpimh_product_sync' );
define( 'PLUGIN_PREFIX', 'wcpimh_' );
define( 'WCPIMH_EC_NAME', 'Holded' );
define( 'WCPIMH_PURCHASE_URL', 'https://checkout.freemius.com/mode/dialog/plugin/5133/plan/8469/' );
// Loads translation.
add_action( 'init', 'wcpimh_load_textdomain' );
/**
 * Load plugin textdomain.
 */
function wcpimh_load_textdomain()
{
    load_plugin_textdomain( 'import-holded-products-woocommerce', false, dirname( plugin_basename( __FILE__ ) ) . '/languages/' );
}

// Includes files.
require_once dirname( __FILE__ ) . '/includes/helpers-functions.php';
require_once dirname( __FILE__ ) . '/includes/class-wcpimh-admin.php';
require_once dirname( __FILE__ ) . '/includes/class-wcpimh-import.php';
require_once dirname( __FILE__ ) . '/includes/update.php';
add_action( 'plugins_loaded', 'imhwc_update_db_check' );
/**
 * Check method updates
 *
 * @return void
 */
function imhwc_update_db_check()
{
    $old_method = get_option( 'wcpimh_api' );
    if ( false !== get_option( 'wcpimh_api' ) ) {
        convert_settings_fields();
    }
    update_option( 'wcpimh_version', WCPIMH_VERSION );
}

add_action( 'admin_notices', 'wcpimh_general_admin_notice' );
/**
 * General Admin Notice.
 *
 * @return void
 */
function wcpimh_general_admin_notice()
{
    $user_id = get_current_user_id();
    
    if ( !get_user_meta( $user_id, 'wcpimh_notice_orders_dismissed' ) ) {
        echo  '<div class="notice notice-warning is-dismissible">' ;
        echo  '<p>' . esc_html__( 'Sync Ecommerce Holded Premium now in version 1.3 brings you to sync Orders after they are completed.', 'import-holded-products-woocommerce' ) . '</p>' ;
        $purchase_url = wp_nonce_url( 'admin.php?page=import_holded-account&wcpimh-notice-orders-dismissed', 'wcpimh_sync_orders' );
        echo  '<p>' . sprintf( '<a href="%s" class="button-primary">%s</a>', $purchase_url, __( 'Purchase Premium Version', 'import-holded-products-woocommerce' ) ) . '</p>' ;
        // phpcs:ignore -- no need to scape
        echo  '</div>' ;
    }

}

add_action( 'admin_init', 'wcpimh_general_admin_notice_dismissed' );
/**
 * Add user meta.
 *
 * @return void
 */
function wcpimh_general_admin_notice_dismissed()
{
    $user_id = get_current_user_id();
    if ( isset( $_GET['wcpimh-notice-orders-dismissed'] ) ) {
        add_user_meta(
            $user_id,
            'wcpimh_notice_orders_dismissed',
            'true',
            true
        );
    }
}


if ( function_exists( 'cmk_fs' ) ) {
    cmk_fs()->set_basename( false, __FILE__ );
} else {
    // DO NOT REMOVE THIS IF, IT IS ESSENTIAL FOR THE `function_exists` CALL ABOVE TO PROPERLY WORK.
    
    if ( !function_exists( 'cmk_fs' ) ) {
        /**
         * Create a helper function for easy SDK access.
         *
         * @return function Dynamic init.
         */
        function cmk_fs()
        {
            global  $cmk_fs ;
            
            if ( !isset( $cmk_fs ) ) {
                // Include Freemius SDK.
                require_once dirname( __FILE__ ) . '/vendor/freemius/wordpress-sdk/start.php';
                $cmk_fs = fs_dynamic_init( array(
                    'id'             => '5133',
                    'slug'           => 'import-holded-products-woocommerce',
                    'premium_slug'   => 'import-holded-woocommerce-premium',
                    'type'           => 'plugin',
                    'public_key'     => 'pk_248ed62e4388ec19335accdf1822c',
                    'is_premium'     => false,
                    'premium_suffix' => '',
                    'has_addons'     => false,
                    'has_paid_plans' => true,
                    'trial'          => array(
                    'days'               => 7,
                    'is_require_payment' => false,
                ),
                    'menu'           => array(
                    'slug'       => 'import_holded',
                    'first-path' => 'admin.php?page=import_holded&tab=settings',
                ),
                    'is_live'        => true,
                ) );
            }
            
            return $cmk_fs;
        }
        
        // Init Freemius.
        cmk_fs();
        // Signal that SDK was initiated.
        do_action( 'cmk_fs_loaded' );
    }

}

register_activation_hook( __FILE__, 'wcpimh_create_db' );
/**
 * Creates the database
 *
 * @since  1.0
 * @access private
 * @return void
 */
function wcpimh_create_db()
{
    global  $wpdb ;
    $charset_collate = $wpdb->get_charset_collate();
    $table_name = $wpdb->prefix . WCPIMH_TABLE_SYNC;
    // DB Tasks.
    $sql = "CREATE TABLE {$table_name} (\n\t    holded_prodid varchar(255) NOT NULL,\n\t    synced boolean,\n          UNIQUE KEY holded_prodid (holded_prodid)\n    \t) {$charset_collate};";
    require_once ABSPATH . 'wp-admin/includes/upgrade.php';
    dbDelta( $sql );
}
